feature/messaging/src/main/kotlin/org/meshtastic/feature/messaging/Message.kt

```diff
@@ -422,25 +426,49 @@ private fun String.ellipsize(maxLength: Int): String = if (length > maxLength) "
  *
  * @param action The [QuickChatAction] to handle.
  * @param messageInputState The [TextFieldState] of the message input field.
+ * @param userPosition Current user position (lat/lng), if available.
  * @param onSendMessage Lambda to call when a message needs to be sent.
  */
 private fun handleQuickChatAction(
     action: QuickChatAction,
     messageInputState: TextFieldState,
+    userPosition: MeshProtos.Position?,
+    ourNode: Node?,
     onSendMessage: (String) -> Unit,
 ) {
+    val processedMessage =
+        if (
+            action.message.contains("%LAT", ignoreCase = true) ||
+            action.message.contains("%LON", ignoreCase = true) ||
+            action.message.contains("%SNAME", ignoreCase = true)
+        ) {
+            var result = action.message
+            userPosition?.let {
+                val latitude = "%.7f".format(it.latitudeI * 1e-7)
+                val longitude = "%.7f".format(it.longitudeI * 1e-7)
+                result =
+                    result.replace("%LAT", latitude, ignoreCase = true).replace("%LON", longitude, ignoreCase = true)
+            }
+            ourNode?.user?.shortName?.let { shortName ->
+                result = result.replace("%SNAME", shortName, ignoreCase = true)
```

Comment:
You're going to have an issue with shortnames with a space, among other cases. 

The space will break your new regex, and other items like `shortName = "<')<"` or `">))>"` _(my poor attempt at a four char ascii art fish)_, will break the Linkify URL parsing. 

You're going to need URI encoding for the label:
```suggestion
                result = result.replace("%SNAME", Uri.encode(shortName), ignoreCase = true)
```

But this URI-encodes all uses of `%SNAME`.

Some options:
* Have a `%SNAMEENC,` which URI encodes
* Falling back to a `%GEO` token which directly get replaced with the full `"geo:(${latitude),${longitude}(${Uri.encode(label)})"`
  * Granted, I like the flexibility of independent replacement tokens, as I want to emit `"geo:47.62148,-122.34814(FriendXYZ+@+2025-11-09+16:40)"`, which would be from hypothetical future tokens of `"geo:%LAT,%LON(%LNAMEENC+@+$DATE+$TIME)"` (timestamp is so folks don't confuse old map pins)
* Handeling on the receiver side -- from `geo:47.62148,-122.34814`, auto-append a label of who+when, if a label doesn't exist, before sending to Linkify

Reply:
I applied your URI encoding change.  (I can't believe I didn't think about that before)
I implemented %LNAME  (but I did not add it to the template, I like the shorter label on maps)
The pin created is ephemeral (unlike waypoint pins) so adding date/timestamps seems unnecessary, as they only exist for the lifetime of the view session. 

In the context of quick chat messages, it seems redundant to want to include an unencode version of the node name in the message, because every message will already have that (sender).

What sort of failure/fallback cases are you thinking about for the simpler %GEO message use?

Reply:
Might I suggest %F0%9F%90%A0 for your fish?  :)

Reply:
> What sort of failure/fallback cases are you thinking about for the simpler %GEO message use?

No failure cases, just mentioning that you can use a `%GEO` token to produce the full message, and you wouldn't have to worry about _when_ to encode `%SNAME`. 

In your updated code, you're always encoding the `%SNAME` & the now added `%LNAME` too. This has the issue of URI encoding when not needed, like with a quick message of `"Test message seen by %LNAME in SOMA"`; sending `%SNAME` or `%LNAME` is useful as _most_ nodes in my list don't have a name, so I assume others likewise don't know my node's name (even more likely for new nodes):
<table><tr><tr><th>Web client view</th><th>Android client view</th></tr><tr><td><img width="250" height="563" alt="Screenshot 2025-11-13 at 07 51 39" src="https://github.com/user-attachments/assets/36fb16fa-249c-47f8-9135-7bd8385da0dd" />No names for most</td><td>
<img width="720" height="996" alt="image" src="https://github.com/user-attachments/assets/ddc27d6b-1e90-4ac3-bf26-9da4da798ca3" /></td></tr></table>
I'm sure there's better use cases for node names that I'm not considering. 
<br/><br/>

> The pin created is ephemeral (unlike waypoint pins) so adding date/timestamps seems unnecessary, as they only exist for the lifetime of the view session, and the messages they appear in already have a timestamp.

The pins can be non-ephemeral once in the map apps, directly so if bookmarked, seen in search history, or used as a routing destination. I certainly have a ton of _old_ bookmarked temporary locations. 

Message timestamps are lost in [store-and-forward](https://meshtastic.org/docs/configuration/module/store-and-forward-module); so if you're disconnected from the mesh when someone sends their location, you'll get the stored message automatically when you reconnect but it will have the current timestamp displayed on an hours old message, implying the sender is at that location now. Granted that should be fixed in store-and-forward.

> Might I suggest %F0%9F%90%A0 for your fish? :)

I could make a case for spiky boi %F0%9F%90%A1.